app/policies/user_policy.rb

```diff
@@ -0,0 +1,35 @@
+class UserPolicy
+  attr_reader :user, :other_user
+
+  def initialize(user, other_user)
+    @user = user
+    @other_user = other_user
+  end
+
+  def new?
+    authorize_role_assignment
+  end
+
+  def create?
+    authorize_role_assignment
+  end
+
+  def permitted_attributes_for_new
+    [:role]
+  end
+
+  private
+
+  def authorize_role_assignment
+    case user.role
+    when 'admin'
+      true
+    when 'coordinator'
+      other_user.role.in? %w[volunteer social_worker]
+    when 'social_worker'
+      other_user.role.in? %w[volunteer]
```

Comment:
I may have misunderstood the requirement. My understanding is that `admin` and `coordinator` roles could invite others, but `social_worker` would need that ability.

Reply:
I believe this is correct. See my comment here: https://github.com/OfficeMomsandDads/scheduler/issues/6#issuecomment-439729042p